presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1002,14 +1009,15 @@ private Partition buildPartitionObject(String queryId, Table table, PartitionUpd
     }
 
     @Override
-    public void createView(ConnectorSession session, SchemaTableName viewName, String viewData, boolean replace)
+    public void createView(ConnectorSession session, SchemaTableName viewName, String viewData, boolean replace, Optional<String> comment)
     {
-        Map<String, String> properties = ImmutableMap.<String, String>builder()
-                .put(TABLE_COMMENT, "Presto View")
-                .put(PRESTO_VIEW_FLAG, "true")
+        Builder<String, String> properties = ImmutableMap.builder();
+        properties.put(PRESTO_VIEW_FLAG, "true")
                 .put(PRESTO_VERSION_NAME, prestoVersion)
-                .put(PRESTO_QUERY_ID_NAME, session.getQueryId())
-                .build();
+                .put(PRESTO_QUERY_ID_NAME, session.getQueryId());
+        if (comment != null && comment.isPresent()) {
```

Comment:
Just make sure comment will never be null. The point of using Optional is to not have null. Remove the `comment != null` here.

Reply:
To keep original behavior when comment is not provided, let's just do
`.put(TABLE_COMMENT, comment.orElse("Presto View"))`

Reply:
The original behavior is not to show any view comment if no comment is provided during CREATE VIEW. When comment is not provided, "Presto View" is actually written into the persistent storage, but the comment node is skipped in SqlFormatter.java when SHOW CREATE VIEW is called. Therefore, we want to remove the default comment; otherwise, "Presto View" would be the default comment when no comment is provided.

Reply:
You are right.